File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/Java/Go/s826491904_comments_removed.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

	scanner.Scan()
	nStr := scanner.Text()
	N, _ := strconv.Atoi(nStr)

	nums := make([]int, N)

	for i := 0; i < N; i++ {
		scanner.Scan()
		numStr := scanner.Text()
		num, _ := strconv.Atoi(numStr)
		nums[i] = num
	}

	if N == 0 {
		fmt.Println(0)
		return
	}

	max := nums[0]
	min := nums[0]

	for _, num := range nums {
		if num > max {
			max = num
		}
		if num < min {
			min = num
		}
	}

	fmt.Println(max - min)
}

// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/mad/Java/Python/s826491904_comments_removed.py
def solve():
    import sys
    input = sys.stdin
    N = int(input.readline())
    max_val, min_val = map(int, input.readline().split())
    for _ in range(1, N):
        in_val = int(input.readline())
        max_val = max(max_val, in_val)
        min_val = min(min_val, in_val)
    print(max_val - min_val)

solve()
# End of Code
